p2p/client.go

```diff
@@ -151,7 +151,8 @@ func (s *p2pServer) GetSignaturesForDocument(ctx *header.ContextHeader, identity
 		// for now going with context.background, once we have a timeout for request
 		// we can use context.Timeout for that
 		count++
-		go s.getSignatureAsync(ctx.Context(), identityService, *doc, client, collaboratorID, in)
+		c, _ := context.WithTimeout(ctx.Context(), s.config.GetP2PConnectionTimeout())
```

Comment:
yes but thats only for connection, this timeout is needed for all p2p calls otherwise the it waits a long time when connection is about to disconnect.